orb.yml

```diff
@@ -46,29 +47,16 @@ commands:
           name: setup honeycomb buildevents and start trace
           command: |
             # set up our working environment and timestamp the trace
-            mkdir -p /tmp/be/bin-linux /tmp/be/bin-linux-arm64 /tmp/be/bin-darwin
-            date +%s > /tmp/be/build_start
+            mkdir -p ~/project/bin/be-linux-x86_64 ~/project/bin/be-linux-aarch64 ~/project/bin/be-darwin-arm64/ /tmp/buildevents/
+            date +%s > /tmp/buildevents/build_start
       - download_be_executables
       - run:
           name: make them executable
-          command: |
-            chmod 755 /tmp/be/bin-linux/buildevents
-            chmod 755 /tmp/be/bin-linux-arm64/buildevents
-            chmod 755 /tmp/be/bin-darwin/buildevents
+          command: chmod 755 ~/project/bin/be*/buildevents
       - save_be_cache
       - run:
           name: report_step
-          command: |
-            # choose the right buildevents binary
-            if uname -a | grep Linux | grep x86_64 > /dev/null 2>&1 ; then
-              export PATH=$PATH:/tmp/be/bin-linux
-            elif uname -a | grep Linux | grep -e arm64 -e aarch64 > /dev/null 2>&1 ; then
-              export PATH=$PATH:/tmp/be/bin-linux-arm64
-            elif uname -a | grep Darwin > /dev/null 2>&1; then
-              export PATH=$PATH:/tmp/be/bin-darwin
-            fi
-            # use that binary to report this step
-            buildevents step $CIRCLE_WORKFLOW_ID setup $(cat /tmp/be/build_start) start_trace
+          command: ~/project/bin/be-$(uname -s | tr '[:upper:]' '[:lower:]')-$(uname -m)/buildevents step $CIRCLE_WORKFLOW_ID setup $(cat /tmp/buildevents/build_start) start_trace
```

Comment:
I wonder if we could shove the full path of the binary into [$BASH_ENV](https://circleci.com/docs/env-vars/#parameters-and-bash-environment) on setup to avoid repeating this lovely monstrosity everywhere.

Reply:
I had a similar thought, and might try it if I have time.

Reply:
oh one question I had - I assume we need to account for (other folks, if not us) sometimes mixing architectures in the same build pipeline? Otherwise we could just grab the right executable for the initial architecture.

Reply:
> I wonder if we could shove the full path of the binary into [$BASH_ENV](https://circleci.com/docs/env-vars/#parameters-and-bash-environment) on setup to avoid repeating this lovely monstrosity everywhere.

So I was looking at this, but I'm not sure we can do it - if I put the binary path into $BASH_ENV on an `aarch64` executor, and a later step runs on `amd64`, isn't it going to fail?

Reply:
Oh of course 😞